server/src/storage_engine/default_engine.rs

```diff
@@ -213,22 +243,24 @@ impl StorageEngine for DefaultEngine {
         Ok(())
     }
 
-    fn create_file(&self, path: String) -> Result<(), EngineError> {
-        self.file_attr_db.db.put(&path, "f")?;
+    fn create_file(&self, path: String, mode: Mode) -> Result<i32, EngineError> {
         let local_file_name = generate_local_file_name(&self.root, &path);
+        let oflag = OFlag::O_CREAT | OFlag::O_EXCL;
+        let fd = fcntl::open(local_file_name.as_str(), oflag, mode)?;
+        self.file_attr_db.db.put(&path, "f")?;
+        let path_nlink = path.clone() + "_nlink";
```

Comment:
Does `nlink` need to delete after calling `delete_file` ?

Reply:
need it